Corrfunc/mocks/DDtheta_mocks.py

```diff
@@ -351,6 +353,391 @@ def DDtheta_mocks(autocorr, nthreads, binfile,
+    (2, 1)
+
+    .. note:: Since the result might change depending on the computer,
+    doctest is skipped for this function.
+    """
+
+    weights1 = None
+    weights2 = None
+    weight_type = None
+
+    try:
+        from Corrfunc._countpairs_mocks import countpairs_theta_mocks as \
+            DDtheta_mocks_extn
+    except ImportError:
+        msg0 = "Could not import the C extension for the angular " \
+              "correlation function for mocks."
+        raise ImportError(msg0)
+
+    import numpy as np
+    from Corrfunc.utils import translate_isa_string_to_enum, fix_ra_dec, \
+        return_file_with_rbins, convert_to_native_endian, process_weights
+    from future.utils import bytes_to_native_str
+    import itertools
+    import time
+
+    if autocorr == 0:
+        if RA2 is None or DEC2 is None:
+            msg1 = "Must pass valid arrays for RA2/DEC2 for " \
+                  "computing cross-correlation."
+            raise ValueError(msg1)
+    else:
+        RA2 = np.empty(1)
+        DEC2 = np.empty(1)
+
+    if link_in_ra is True:
+        link_in_dec = True
+
+    if link_in_dec is False and link_in_ra is False:
+        msg2 = "Warning: Brute-force calculation without any gridding " \
+               "is forced, as link_in_dec and link_in_ra are both set " \
+               "to False. Please be sure to turn on at least link_in_dec, " \
+               "or both link_in_dec and link_in_ra on."
+        raise ValueError(msg2)
+
+    if link_in_dec is True:
+        if link_in_ra is True:
+
+            weights1, weights2 = \
+                process_weights(weights1, weights2, RA1, RA2, weight_type, autocorr)
+
+            # Ensure all input arrays are native endian
+            RA1, DEC1, weights1, RA2, DEC2, weights2 = [
+                convert_to_native_endian(arr, warn=True) for arr in
+                [RA1, DEC1, weights1, RA2, DEC2, weights2]]
+
+            fix_ra_dec(RA1, DEC1)
+            if autocorr == 0:
+                fix_ra_dec(RA2, DEC2)
+
+            # Passing None parameters breaks the parsing code, so avoid this
+            kwargs = {}
+            for k in ['weights1', 'weights2', 'weight_type', 'RA2', 'DEC2']:
+                v = locals()[k]
+                if v is not None:
+                    kwargs[k] = v
+
+            integer_isa = translate_isa_string_to_enum(isa)
+            rbinfile, delete_after_use = return_file_with_rbins(binfile)
+            bin_refs = np.arange(1, maxbinref + 1)
+            bin_ref_perms = itertools.product(bin_refs, bin_refs)
+            dtype = np.dtype([(bytes_to_native_str(b'nRA'), np.int),
+                              (bytes_to_native_str(b'nDEC'), np.int),
+                              (bytes_to_native_str(b'avg_time'), np.float),
+                              (bytes_to_native_str(b'sigma_time'), np.float)])
+            all_runtimes = np.zeros(maxbinref ** 3, dtype=dtype)
+            all_runtimes[:] = np.inf
+
```

Comment:
All of these lines should precede any looping. You are setting up the whole harness for running with the appropriate values, and those values will be relevant for both cases of ``(link_in_ra==True, link_in_ra==False)``

*edited* For ``link_in_ra==False``, you will have to change `bin_ref_perms`. Something like `[(x, None) for x in bin_refs]` might work. That might eliminate the need for the `if link_in_ra` and the corresponding `else` conditions. (I am not sure if `DDtheta_mocks` allows for `None` for `nRA` when `link_in_ra==False` but worth a shot though - would make for a lot cleaner code)

Reply:
I tried, but `DDtheta_mocks` does not seem to allow `None` for `nRA` when `ink_in_ra==False`.

Reply:
What is the error message returned? (It must be in the python-bindings, because in the actual implementation, `nRA` can have an arbitrary value when `link_in_ra==False`)

Reply:
When I tried to change the `nRA` to `None`, I get the error with traceback:
```
---------------------------------------------------------------------------
error                                     Traceback (most recent call last)
error: ArgumentError: In DDtheta_mocks> Could not parse the arguments. Input parameters are: 
autocorr, nthreads, binfile, RA1, DEC1, weights1, RA2, DEC2, weights2, link_in_dec, link_in_ra, verbose, output_thetaavg, fast_acos, ra_refine_factor, dec_refine_factor, max_cells_per_dim, copy_particles, enable_min_sep_opt, c_api_timer, isa, weight_type, 
RuntimeError occurred with perms = (1, 1)
Continuing...
```

However, toward the end of the output, I also got the following result: 
``` python
((1, 1),
 array([(                   1,                    1,  0.,  0.),
        (                   1,                    2,  0.,  0.),
        (                   1,                    3,  0.,  0.),
        (                   2,                    1,  0.,  0.),
        (                   2,                    2,  0.,  0.),
        (                   2,                    3,  0.,  0.),
        (                   3,                    1,  0.,  0.),
        (                   3,                    2,  0.,  0.),
        (                   3,                    3,  0.,  0.),
        (-9223372036854775808, -9223372036854775808, inf, inf),
        (-9223372036854775808, -9223372036854775808, inf, inf),
        (-9223372036854775808, -9223372036854775808, inf, inf),
        (-9223372036854775808, -9223372036854775808, inf, inf),
        (-9223372036854775808, -9223372036854775808, inf, inf),
        (-9223372036854775808, -9223372036854775808, inf, inf),
        (-9223372036854775808, -9223372036854775808, inf, inf),
        (-9223372036854775808, -9223372036854775808, inf, inf),
        (-9223372036854775808, -9223372036854775808, inf, inf),
        (-9223372036854775808, -9223372036854775808, inf, inf),
        (-9223372036854775808, -9223372036854775808, inf, inf),
        (-9223372036854775808, -9223372036854775808, inf, inf),
        (-9223372036854775808, -9223372036854775808, inf, inf),
        (-9223372036854775808, -9223372036854775808, inf, inf),
        (-9223372036854775808, -9223372036854775808, inf, inf),
        (-9223372036854775808, -9223372036854775808, inf, inf),
        (-9223372036854775808, -9223372036854775808, inf, inf),
        (-9223372036854775808, -9223372036854775808, inf, inf)],
       dtype=[('nRA', '<i8'), ('nDEC', '<i8'), ('avg_time', '<f8'), ('sigma_time', '<f8')]))
```
which does not seem very meaningful.